src/mmw/apps/modeling/views.py

```diff
@@ -254,22 +256,29 @@ def start_mapshed(request, format=None):
 
 
 def _initiate_mapshed_job_chain(mapshed_input, job_id):
+    workers = choose_workers()
+    get_worker = lambda: random.choice(workers)
```

Comment:
Pooling the workers for each job made the biggest difference, but I did notice a small change just from splitting the task up as well, although I don't have as clear an idea on why exactly that might be the case. I think that the former implementation would be equivalent with enough workers or higher degrees of concurrency, but I'm similarly unsure about what happens in the even of failure in each case.